tldw_Server_API/app/core/Sharing/unified_share_audit.py

```
@@ -0,0 +1,315 @@
+                tenant_user_id,
+                context_user_id,
+                context_ip_address,
+                context_user_agent,
+                resource_type,
+                resource_id,
+                metadata
+            FROM audit_events
+            WHERE {' AND '.join(conditions)}
+            ORDER BY timestamp DESC, event_id DESC
+            LIMIT ? OFFSET ?
+        """
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)